packages/tunnel/src/commands/deploy/utils.ts

```diff
@@ -0,0 +1,161 @@
+import { appendPath } from '@silverhand/essentials';
+import AdmZip from 'adm-zip';
+import chalk from 'chalk';
+import ora from 'ora';
+
+import { consoleLog } from '../../utils.js';
+
+type TokenResponse = {
+  access_token: string;
+  token_type: string;
+  expires_in: number;
+  scope: string;
+};
+
+type DeployArgs = {
+  auth: string;
+  endpoint: string;
+  experiencePath: string;
+  managementApiResource?: string;
+  verbose: boolean;
+};
+
+export const deployToLogtoCloud = async ({
+  auth,
+  endpoint,
+  experiencePath,
+  managementApiResource,
+  verbose,
+}: DeployArgs) => {
+  const spinner = ora();
+  if (verbose) {
+    spinner.start('[1/4] Zipping files...\n');
+  }
+  const zipBuffer = await zipFiles(experiencePath);
+  if (verbose) {
+    spinner.succeed();
+  }
+
+  try {
+    if (verbose) {
+      spinner.start('[2/4] Exchanging access token...\n');
+    }
+    const endpointUrl = new URL(endpoint);
+    const tokenResponse = await getAccessToken(auth, endpointUrl, managementApiResource);
+    if (verbose) {
+      spinner.succeed();
```

Comment:
nit: it will still look good without these empty lines i think

Reply:
The `spinner.suceed()` actually marks the spinning spinner as a checkmark without changing the text message, I think. But now I think it's probably even better to append a `Done.` at the end of the original spinner message. What do you think?
```
spinner.start('Uploading zip...');
...
spinner.succeed('Uploading zip... Done.')
```